java/ql/src/experimental/Security/CWE/CWE-093/CrlfInjection.qhelp

```diff
@@ -0,0 +1,39 @@
+
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+
+<p>If unprocessed user input is directly added to email headers or written to logs, malicious users can carry out email or 
```

Comment:
Does this query have anything to do with log injection?

Reply:
Log injection, http header injection, and email header injection all belong to CRLF injection, so consider adding log injection.

Reply:
Please take a look at whether the current qhelp meets expectations, thank you!

Reply:
It does not. You have written a query that checks for CRLF injection into email subjects, then written qhelp that talks about lots of other places where a rogue CRLF is a problem that are otherwise unrelated to this query.

Just describe what vulnerability this query targets: CRLF injection into emails.

Reply:
Do I need to delete the sink and related documents injected by the log?

Reply:
Aha, apologies I had not noticed

```
    or
    exists(LoggingCall c | c.getALogArgument() = this.asExpr())
```

This makes more sense now. Even so you should mention just what the query does.

Reply:
Can `overview` not describe the purpose of this query?

Reply:
Forget it, I'll make some more precise suggestions later in the review process. For now let's wait for security lab to make their evaluation.

Reply:
Well, I am very sorry for that.